lark/utils.py

```diff
@@ -12,6 +13,17 @@
 # By default, we should not output any log messages
 logger.setLevel(logging.CRITICAL)
 
+def isalnum(x):
```

Comment:
~~Why are we behaving differently than `str.isalnum`?~~

~~I know that you didn't write this first, but I think that we should still add a comment explaining this.~~

Ok. I found it. This method has the wrong name. It should be named `is_id_continue`. (and should still have an explaining comment)

Reply:
Hello,
I had named it `isalnum` because it is the unicode-compatible version of `str.isalnum`. But I changed the name now to avoid confusion.

Reply:
It isn't. That is my point. Please add a reference to PEP 3131, and also change the name of `isalpha` to `is_id_continue`.

Also, I think there is a better way of going about this somehow using the information that python internally uses, but I am not sure what it is.

Reply:
Hello @MegaIng : In that case the distinction eludes me (if you can share some pointers please do so!). I have added the reference to PEP 3131. The problem with renaming `isalpha` is that `isalphanum` has just been renamed to `is_id_continue`, and both serve different purposes (`isalpha` should match only `id_start` characters, and not numeric characters).

Reply:
Read PEP 3131. The `str.isalphanum` method is already unicode compatible. But this list of unicode categories includes a few additional things that are valid inside python identifiers, but aren't really alphabetic. As example 'connector punctuations (Pc)' or the underscore.

Reply:
Thank you! When I said "unicode compatible" I think I meant "international compatible". Since `isalpha` is, in practice here, used to detect characters that make up a word, and `isalphanum` the same but including numerals, the goal was to make a function that behaved the same no matter which human language or script the word belongs to.

That said, I am not sure anymore that `is_id_continue` is a good name, because it leads to suppose that `_` would pass, when it does not (and should not, as the original `'_'.isalpha()` returns `False`).

Anyways, the current functionality is correct, and as this is simply a question of the best name for the function I will wait for your final recommendations.

Reply:
I think `_` should pass. And sorry, I mistyped, `is_id_start` was what it was supposed to be.

Reply:
Done! Thank you.